lib/manager/components/DeploymentViewer.js

```diff
@@ -447,85 +565,7 @@ export default class DeploymentViewer extends Component<Props, State> {
-                    options={options ? options.map(v => ({value: v, label: v})) : []} />
-                </ListGroupItem>
-                <ListGroupItem>
-                  Deploying to the{' '}
-                  <small><em>{deployment.routerId || 'default'}</em></small>{' '}
-                  OpenTripPlanner router.
-                </ListGroupItem>
-                <ListGroupItem>
-                  OpenStreetMap Settings
-                  <Checkbox
-                    checked={!deployment.skipOsmExtract}
-                    onChange={this._onChangeSkipOsmExtract}>
-                    Build graph with OSM extract
-                  </Checkbox>
-                  {/* Hide URL/auto-extract if skipping OSM extract. */}
-                  {deployment.skipOsmExtract
-                    ? null
-                    : deployment.osmExtractUrl
-                      ? <div>
-                        <ControlLabel className='unselectable buffer-right'>
-                          URL:
-                        </ControlLabel>
-                        <span
-                          className='overflow'
-                          style={{
-                            marginBottom: '5px',
-                            width: '240px',
-                            verticalAlign: 'middle'
-                          }}
-                          title={deployment.osmExtractUrl}>
-                          {deployment.osmExtractUrl}
-                        </span>
-                        <Button
-                          bsSize='xsmall'
-                          style={{marginLeft: '5px'}}
-                          onClick={this._setOsmUrl}>Change</Button>
-                        <Button
-                          bsSize='xsmall'
-                          style={{marginLeft: '5px'}}
-                          onClick={this._clearOsmUrl}>Clear</Button>
-                      </div>
-                      : <div>
-                        Auto-extract OSM (N. America only)
-                        <Button
-                          bsSize='xsmall'
-                          style={{marginLeft: '5px'}}
-                          onClick={this._setOsmUrl}>Override</Button>
-                      </div>
-                  }
-                </ListGroupItem>
-                <ListGroupItem>
-                  <CustomConfig
-                    name='customBuildConfig'
-                    label='Build'
-                    updateDeployment={updateDeployment}
-                    deployment={deployment} />
-                </ListGroupItem>
-                <ListGroupItem>
-                  <CustomConfig
-                    name='customRouterConfig'
-                    label='Router'
-                    updateDeployment={updateDeployment}
-                    deployment={deployment} />
-                </ListGroupItem>
-              </ListGroup>
-            </Panel>
+            {this.renderConfigurationsPanel()}
```

Comment:
Given that this file is so long already, I think my preferred refactor would have been to split out the components into smaller files, but this approach is fine for now.